applications/FluidDynamicsApplication/tests/CompressibleSymbolicGeneration/compilable_symbolic_test.h

```diff
@@ -0,0 +1,312 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Eduard Gómez
+//
+
+#include <array>
+#include <iomanip>
+#include <cmath>
+#include <ostream>
+#include <iostream>
+
+/**
+ * This namespace contains classes and function mimicking the necessary
+ * behaviour of Kratos in order to run the generated code without having
+ * to modify it.
+ */
+namespace TestCompressibleNavierStokesSymbolic {
+
+/** Matrix-like class with only:
+ * - storage manipulation access
+ * - operator<<
+ * - almost equality check
+ */
+template<std::size_t TRows, std::size_t TCols>
+class Matrix
```

Comment:
Why you require your own matrix class? Can't you use Kratos bounded one?

Reply:
Two reasons:
- The Kratos source is not packaged with the Python package. This means that when running the test, the location of the kratos source is unknown. It may not even be in the machine at all if Kratos was installed with pip.
- Compilation time is kept almost instantaneous.

For these two reasons the cpp code in this test is completely isolated from Kratos.